### PR TITLE
 Use, filter on and create order labels

### PR DESCRIPTION
**Pull request by ithiame**
#1415 add label in order

### CODE REVIEW

File: src/routes/(app)/admin[[hash=admin_hash]]/label/[id]/+page.svelte [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	function confirmDelete(event: Event) {
+		if (!confirm('Would you like to delete this Contact form?')) {
+			event.preventDefault();
+		}
+	}
+</script>
+
+<form method="post" class="flex flex-col gap-4">
+	<label class="form-label">
+		name
```
Comment (coyotte508) [inline src/routes/(app)/admin[[hash=admin_hash]]/label/[id]/+page.svelte:18]:
```suggestion
		Name
```

File: src/routes/(app)/admin[[hash=admin_hash]]/label/[id]/contact-form-schema.ts
Comment (coyotte508) [inline src/routes/(app)/admin[[hash=admin_hash]]/label/[id]/contact-form-schema.ts:1]: Why is this here? There's no contact form in this PR right?

Comment (ithiame) [reply src/routes/(app)/admin[[hash=admin_hash]]/label/[id]/contact-form-schema.ts:1]: Yes. It was a mistake

File: src/routes/(app)/admin[[hash=admin_hash]]/order/[id]/label/+page.server.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	if (!order) {
+		throw error(404, 'Order not found');
+	}
+	return {
+		labels,
+		orderLabelIds: order.orderLabelIds
+	};
+};
+export const actions = {
+	default: async function ({ request, params }) {
```
Comment (coyotte508) [inline src/routes/(app)/admin[[hash=admin_hash]]/order/[id]/label/+page.server.ts:25]: We should also check that the order label ids given are in DB

File: src/lib/types/Order.ts [hunk 3]
```diff
@@ -255,6 +256,7 @@ export interface Order extends Timestamps {
 		acceptedDepositConditionsAndFullPayment?: boolean;
 		acceptedExportationAndVATObligation?: boolean;
 	};
+	orderLabelIds?: OrderLabel['_id'][];
```
Comment (coyotte508) [inline src/lib/types/Order.ts:259]: A DB index should be added to `orderLabelIds`

**Review by coyotte508 — COMMENTED** [review]
With the requested changes, should be good to merge!

Nice PR overall

File: src/routes/(app)/admin[[hash=admin_hash]]/order/[id]/label/+page.svelte [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+		<MultiSelect
+			options={data.labels.map((label) => ({
+				value: label._id,
+				label: label.name
+			}))}
+			bind:selected={selectedLabel}
+		/>
+	</label>
+	{#each selectedLabel.map((label) => label.value) as labelId, i}
+		<input type="hidden" name="orderLabelId[{i}]" value={labelId} />
```
Comment (coyotte508) [inline src/routes/(app)/admin[[hash=admin_hash]]/order/[id]/label/+page.svelte:26]:
```suggestion
		<input type="hidden" name="orderLabelIds[{i}]" value={labelId} />
```

File: src/routes/(app)/admin[[hash=admin_hash]]/label/[id]/+page.server.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+				}
+			}
+		);
+	},
+
+	delete: async function ({ params }) {
+		await collections.labels.deleteOne({
+			_id: params.id
+		});
+		await collections.orders.updateMany({}, { $pull: { orderLabelIds: params.id } });
```
Comment (coyotte508) [inline src/routes/(app)/admin[[hash=admin_hash]]/label/[id]/+page.server.ts:58]:
```suggestion
		await collections.orders.updateMany({orderLabelIds: params.id}, { $pull: { orderLabelIds: params.id } });
```

To use the index

File: src/lib/server/database.ts [hunk 6]
```diff
@@ -139,6 +141,7 @@ const indexes: Array<[Collection<any>, IndexSpecification, CreateIndexesOptions?
 	[collections.orders, { 'payments.invoice.number': 1 }, { unique: true, sparse: true }],
 	[collections.orders, { 'payments.status': 1 }],
 	[collections.orders, { status: 1, 'payments.status': 1 }],
+	[collections.orders, { orderLabelIds: 1 }],
```
Comment (coyotte508) [inline src/lib/server/database.ts:144]:
```suggestion
	[collections.orders, { orderLabelIds: 1 }, {sparse: true}],
```

**Review by coyotte508 — APPROVED** [review]

**Comment by Tirodem** [comment]
I'm quite proud.
It has "de la gueule".
image
image
image
As we need to go on "(+)" to edit and remove a label, one day, we should edit the icon for a ✏ for edit or something like that, but it's really nice.

**Pull request merged**